src/main/scala/PropertyChecker.scala

```diff
@@ -385,7 +385,31 @@ object PropertyChecker {
             }
           }
         } else if (PropertyCheckerConstants.NumericFormatDataTypes.contains(baseValue)) {
-          throw new NotImplementedError() // Implement after adding NumberFormat class
+          var format = objectNode.get("format")
```

Comment:
Yeah, let's start breaking this function up. It's a monster. Can you start by moving this bit you've added into its own function?

Reply:
Done